ludwig/hyperopt/execution.py

```diff
@@ -585,6 +588,101 @@ def execute(
+        hyperopt_dict["config"] = modified_config
+        hyperopt_dict["experiment_name"] = f'{hyperopt_dict["experiment_name"]}_{trial_id}'
+
+        train_stats, eval_stats = run_experiment(**hyperopt_dict)
+        metric_score = self.get_metric_score(eval_stats)
+
+        tune.report(parameters=str(config), metric_score=metric_score,
+                    training_stats=str(train_stats), eval_stats=str(eval_stats))
+
+    def execute(self,
+                config,
+                dataset=None,
+                training_set=None,
+                validation_set=None,
+                test_set=None,
+                training_set_metadata=None,
+                data_format=None,
+                experiment_name="hyperopt",
+                model_name="run",
+                # model_load_path=None,
+                # model_resume_path=None,
+                skip_save_training_description=False,
+                skip_save_training_statistics=False,
+                skip_save_model=False,
+                skip_save_progress=False,
+                skip_save_log=False,
+                skip_save_processed_input=False,
+                skip_save_unprocessed_output=False,
+                skip_save_predictions=False,
+                skip_save_eval_stats=False,
+                output_directory="results",
+                gpus=None,
+                gpu_memory_limit=None,
+                allow_parallel_threads=True,
+                use_horovod=None,
+                random_seed=default_random_seed,
+                debug=False,
+                **kwargs):
+
+        hyperopt_dict = dict(
+            config=config,
+            dataset=dataset,
+            training_set=training_set,
+            validation_set=validation_set,
+            test_set=test_set,
+            training_set_metadata=training_set_metadata,
+            data_format=data_format,
+            experiment_name=experiment_name,
+            model_name=model_name,
+            # model_load_path=model_load_path,
+            # model_resume_path=model_resume_path,
+            eval_split=self.split,
+            skip_save_training_description=skip_save_training_description,
+            skip_save_training_statistics=skip_save_training_statistics,
+            skip_save_model=skip_save_model,
+            skip_save_progress=skip_save_progress,
+            skip_save_log=skip_save_log,
+            skip_save_processed_input=skip_save_processed_input,
+            skip_save_unprocessed_output=skip_save_unprocessed_output,
+            skip_save_predictions=skip_save_predictions,
+            skip_save_eval_stats=skip_save_eval_stats,
+            output_directory=output_directory,
+            gpus=gpus,
+            gpu_memory_limit=gpu_memory_limit,
+            allow_parallel_threads=allow_parallel_threads,
+            use_horovod=use_horovod,
+            random_seed=random_seed,
+            debug=debug,
+        )
+
+        analysis = tune.run(tune.with_parameters(
```

Comment:
are there parameters for configuring the number of evaluations (i.e., num_samples)? 
Also, it'd be nice to set the `resources_per_trial` of `tune.run`. This would allow Tune to allocate GPUs correctly. 

Finally, I'm interested in this `use_horovod` mechanism; what would it take to get that working?

Reply:
We're going to be implementing the `use_horovod` support in a follow-up PR.  As part of the Dask / Ray integration, we're changing this API so that `use_horovod` gets replaced with a `Backend` interface.

Reply:
Sounds good - yeah, i think this entire comment can be left to future work.

Reply:
> are there parameters for configuring the number of evaluations (i.e., num_samples)?

They are actually in the sampler rather than in the exectuor. So far we tried to keep the two consepts separated.

Reply:
> are there parameters for configuring the number of evaluations (i.e., num_samples)?

Yes, I have added it to the sampler. The `sampler` will be returning the `search_space` and `num_samples`. This way, we could have the `sampler` and `executor` decoupled.

Reply:
@richardliaw I have a question regarding Resource Allocation for `resources_per_trial`. We get the parameters `gpus` and `gpu_memory_limit` from the User. `gpus` is GPUs to use (it uses the same syntax of CUDA_VISIBLE_DEVICES) and `gpu_memory_limit` is the maximum memory in MB to allocate per GPU device. 
Is there a way to allocate resources using `gpu_id` (CUDA_VISIBLE_DEVICES Syntax) for `tune.run`?Also, these params will be further passed to the experiment for the [Initialize Tensorflow](https://github.com/uber/ludwig/blob/master/ludwig/utils/tf_utils.py#L48) functionality.